structure/deque/deque_test.go

```diff
@@ -0,0 +1,407 @@
+//
+// Package deque provides a double-ended queue data structure implementation.
```

Comment:
Do you need to repeat yourself again here? I think not 😄 
```suggestion
```

Reply:
Yeah true, lemme edit it quickly!